Lib/glyphsLib/builder/builders.py

```diff
@@ -90,6 +90,12 @@ def masters(self):
             return self._ufos.values()
         kerning_groups = {}
 
+        # Store list of actually existing master (layer) ids. This helps with
+        # catching dangling layer data that Glyphs may ignore, e.g. when
+        # copying glyphs from other fonts with, naturally, different master
+        # ids.
+        master_layer_ids = [m.id for m in self.font.masters]
```

Comment:
since you don't care about the order but use this just for checking if something is inside it, you should use a set rather than a list `{m.id for m in self.font.masters}`.
I think (hope) that something that is called `id` is unique...

Reply:
https://github.com/schriftgestalt/GlyphsSDK/blob/master/GlyphsFileFormat.md says: "id string a unique id that connects the layers (associated ID) with the master."